### PR TITLE
enhancement for addon management

### PR DESCRIPTION
**Pull request by skeeey**
Signed-off-by: liuwei <liuweixa@redhat.com>

### CODE REVIEW

**Comment by skeeey** [comment]
refer to https://github.com/open-cluster-management/enhancements/issues/12

File: enhancements/sig-architecture/12-addon-manager/README.md [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+- For each addon, if it needs to register to hub cluster, it needs to leverage addon framework to
+  recode its registration configuration information to the its `ManagedClusterAddOn` CR
+
+### Version Skew Strategy
+
+This design will be implemented on ACM 2.3
+
+For addon registration part, it is a new feature, it will not affect the addons of the old version.
+
+For lease update part, it will keep compatibility with 2.2, and after 2.3, the compatibility of 2.2
```
Comment (qiujian16) [inline enhancements/sig-architecture/12-addon-manager/README.md:299]: one thing we have not considered is the new hub with old  klusterlet/old agent. In this case, the addon status will be unavailable?

Comment (skeeey) [reply enhancements/sig-architecture/12-addon-manager/README.md:299]: there will be no available status for addons

File: enhancements/sig-architecture/12-addon-manager/README.md [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ManagedClusterAddOn
+metadata:
+  name: observability
+  namespace: cluster1
+spec:
+  installNamespace: open-cluster-management-addon-observability
+  registrations:
```
[the remark this answers is not in the document]
Comment (skeeey) [reply enhancements/sig-architecture/12-addon-manager/README.md:237]: updated

File: enhancements/sig-architecture/12-addon-manager/README.md [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+  - signerName: "kubernetes.io/kube-apiserver-client"
+  - signerName: "observability-signer"
```
[the remark this answers is not in the document]
Comment (skeeey) [reply enhancements/sig-architecture/12-addon-manager/README.md:239]: updated

File: enhancements/sig-architecture/12-addon-manager/README.md [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+
+- The `ManagedClusterAddOn` API should be updated on the hub cluster
+- For lease update, each addon needs to create and update its lease on its installation namespace
+  on the managed cluster
+- For each addon, if it needs to register to hub cluster, it needs to leverage addon framework to
+  recode its registration configuration information to the its `ManagedClusterAddOn` CR
+
+### Version Skew Strategy
+
+This design will be implemented on ACM 2.3
```
[the remark this answers is not in the document]
Comment (skeeey) [reply enhancements/sig-architecture/12-addon-manager/README.md:295]: removed

File: enhancements/sig-architecture/12-addon-manager/README.md [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+
+The registration agent maintains the status of managed cluster on the hub cluster, so the registration
+hub controller watches the managed cluster status, if the status of managed cluster is `unknown`,
+the registration hub controller will update the status of all addons to `unknown` on the hub cluster.
+
+By this way, we can eliminate the addon lease update requests from managed clusters to hub cluster.
+
+### Addon registration
+
+![addon registration](addon-registration.png)
```
[the remark this answers is not in the document]
Comment (skeeey) [reply enhancements/sig-architecture/12-addon-manager/README.md:63]: removed

File: enhancements/sig-architecture/12-addon-manager/README.md [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
+
+We want to follow up the [addon framework proposal](https://github.com/open-cluster-management/enhancements/issues/8)
+to use open-cluster-management registration to manage the lease update and registration of all
+managed cluster addons.
+
+## Design Details
+
+### Lease update
+
+![lease update](lease-update.png)
```
[the remark this answers is not in the document]
Comment (skeeey) [reply enhancements/sig-architecture/12-addon-manager/README.md:48]: removed the pic

File: enhancements/sig-architecture/12-addon-manager/README.md [hunk 7]
```diff
@@ -0,0 +1,300 @@
...
+
+- Maintain addon status to available if addon update its lease constantly
+- Update addon status to unavailable if addon stops to update its lease
+- Update addon status to unknown if there is no lease for this addon
+- Update addon status to unknown if managed cluster becomes to unknown
+
+### Graduation Criteria
+
+#### GA Graduation
+
```
[the remark this answers is not in the document]
Comment (skeeey) [reply enhancements/sig-architecture/12-addon-manager/README.md:280]: added

File: enhancements/sig-architecture/12-addon-manager/README.md [hunk 8]
```diff
@@ -0,0 +1,300 @@
...
+      groups:
+      - open-cluster-management-observability
+      extras:
+        organizationalUnit: acm
+```
+
+In this example, the observability addon requires two csrs. One to access hub kube-api (with singer name
+`kubernetes.io/kube-apiserver-client`), and another to access its own endpoint (with singer name
+`observability-signer`). It depends on its own hub addon controller to decide whether the csr
+should be approved or not.
```
[the remark this answers is not in the document]
Comment (skeeey) [reply enhancements/sig-architecture/12-addon-manager/README.md:251]: added

**Pull request merged**